logos-derive/src/lib.rs

```diff
@@ -132,14 +132,35 @@ pub fn logos(input: TokenStream) -> TokenStream {
                         }
                     };
 
-                    let bytes = definition.literal.to_bytes();
-                    let then = graph.push(
-                        leaf(definition.literal.span())
-                            .priority(definition.priority.unwrap_or(bytes.len() * 2))
-                            .callback(definition.callback),
-                    );
-
-                    ropes.push(Rope::new(bytes, then));
+                    if definition.ignore_flags.is_empty() {
+                        let bytes = definition.literal.to_bytes();
+                        let then = graph.push(
+                            leaf(definition.literal.span())
+                                .priority(definition.priority.unwrap_or(bytes.len() * 2))
+                                .callback(definition.callback),
+                        );
+
+                        ropes.push(Rope::new(bytes, then));
+                    } else {
+                        let mir = definition
+                            .literal
+                            .escape_regex()
```

Comment:
Is this necessary?

Reply:
Well, as I'm calling `.to_mir` right after, we need to ensure the literal will actually be used as a string and not some regex semantics.

For example, `#[token("[Some Header]", ignore(case))]` needs to be properly escaped so that it does not turn into `Mir::Alternation('S' ... 'r')` but `Mir::Concat('[', 'S', ... , 'r', ']')`.

Originally, I had written a function that directly converts any literal into a case insensitive MIR but I later found out about the `case_insensitive` function on HIR's `ParserBuilder`. For this reason, I thought it preferable to put everything into the `to_mir` function.